VocaDb.Migrations/Migrations.cs

```diff
@@ -1,10 +1,95 @@
 using System.Data;
+using System.Linq;
 using FluentMigrator;
 
 namespace VocaDb.Migrations
 {
 	// Migration version format: YYYY_MM_DD_HHmm
 
+	[Migration(2020_11_20_2000)]
+	public class Comments : Migration
+	{
+		private sealed class CommentTable
+		{
+			public CommentTable(string name, string schema, bool hasAuthorName)
+			{
+				Name = name;
+				Schema = schema;
+				HasAuthorName = hasAuthorName;
+			}
+
+			public string Name { get; }
+			public string Schema { get; }
+			public bool HasAuthorName { get; }
+
+			public string NameWithSchema => Schema != null ? $"{Schema}.{Name}" : Name;
+		}
+
+		public override void Up()
+		{
+			Rename.Column("Date").OnTable(TableNames.AlbumReviews).To("Created");
+			Rename.Column("Text").OnTable(TableNames.AlbumReviews).To("Message");
+			Rename.Column("User").OnTable(TableNames.AlbumReviews).To("Author");
+			Delete.Index("UX_AlbumReviews").OnTable(TableNames.AlbumReviews);
+
+			var commentTables = new[]
+			{
+				new CommentTable(name: TableNames.AlbumComments, schema: null, hasAuthorName: true),
+				new CommentTable(name: TableNames.ArtistComments, schema: null, hasAuthorName: false),
+				new CommentTable(name: TableNames.DiscussionComments, schema: "discussions", hasAuthorName: true),
+				new CommentTable(name: TableNames.ReleaseEventComments, schema: null, hasAuthorName: false),
+				new CommentTable(name: TableNames.SongComments, schema: null, hasAuthorName: false),
+				new CommentTable(name: TableNames.SongListComments, schema: null, hasAuthorName: false),
+				new CommentTable(name: TableNames.TagComments, schema: null, hasAuthorName: false),
+				new CommentTable(name: TableNames.UserComments, schema: null, hasAuthorName: false),
+
+				new CommentTable(name: TableNames.AlbumReviews, schema: null, hasAuthorName: false),
+			};
+
+			Create.Table(TableNames.Comments)
+				.WithColumn("Id").AsInt64().NotNullable().Identity().PrimaryKey()
+				.WithColumn("Author").AsInt32().NotNullable().ForeignKey(TableNames.Users, "Id").OnDelete(Rule.Cascade)
+				.WithColumn("Created").AsDateTime().NotNullable()
+				.WithColumn("Deleted").AsBoolean().NotNullable().WithDefaultValue(false)
+				.WithColumn("Message").AsString(4000).NotNullable()
+				// TODO: remove these columns
+				.WithColumn("OldTable").AsString().Nullable()
+				.WithColumn("OldId").AsInt32().Nullable();
+
+			var values = string.Join(" union ", commentTables.Select(table => $"select '{table.NameWithSchema}', Id, Author, Created, Message from {table.NameWithSchema}"));
+			Execute.Sql($@"insert into Comments(OldTable, OldId, Author, Created, Message) {values} order by Created");
+
+			foreach (var table in commentTables)
+			{
+				Create.Column("Comment").OnTable(table.Name).InSchema(table.Schema).AsInt64().Nullable();

```

Comment:
Should this be marked a primary key? And unique most likely. I know you're still editing this...